// @material-ui/lab/internal/pickers/wrappers/StaticWrapper.js
import * as React from 'react';
import { withStyles } from '@material-ui/core/styles';
import { DIALOG_WIDTH } from '../constants/dimensions';
import { WrapperVariantContext, IsStaticVariantContext } from './WrapperVariantContext';
import { jsx as _jsx } from "react/jsx-runtime";

const styles = theme => ({
  root: {
    overflow: 'hidden',
    minWidth: DIALOG_WIDTH,
    display: 'flex',
    flexDirection: 'column',
    backgroundColor: theme.palette.background.paper
  }
});

const StaticWrapper = props => {
  const {
    classes,
    displayStaticWrapperAs = 'mobile',
    children
  } = props;
  const isStatic = true;
  return /*#__PURE__*/_jsx(IsStaticVariantContext.Provider, {
    value: isStatic,
    children: /*#__PURE__*/_jsx(WrapperVariantContext.Provider, {
      value: displayStaticWrapperAs,
      children: /*#__PURE__*/_jsx("div", {
        className: classes.root,
        children: children
      })
    })
  });
};

export default withStyles(styles, {
  name: 'MuiPickersStaticWrapper'
})(StaticWrapper);
