// @material-ui/lab/node/internal/pickers/wrappers/StaticWrapper.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _styles = require("@material-ui/core/styles");

var _dimensions = require("../constants/dimensions");

var _WrapperVariantContext = require("./WrapperVariantContext");

var _jsxRuntime = require("react/jsx-runtime");

const styles = theme => ({
  root: {
    overflow: 'hidden',
    minWidth: _dimensions.DIALOG_WIDTH,
    display: 'flex',
    flexDirection: 'column',
    backgroundColor: theme.palette.background.paper
  }
});

const StaticWrapper = props => {
  const {
    classes,
    displayStaticWrapperAs = 'mobile',
    children
  } = props;
  const isStatic = true;
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_WrapperVariantContext.IsStaticVariantContext.Provider, {
    value: isStatic,
    children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_WrapperVariantContext.WrapperVariantContext.Provider, {
      value: displayStaticWrapperAs,
      children: /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
        className: classes.root,
        children: children
      })
    })
  });
};

var _default = (0, _styles.withStyles)(styles, {
  name: 'MuiPickersStaticWrapper'
})(StaticWrapper);

exports.default = _default;
