// @material-ui/lab/legacy/internal/pickers/wrappers/StaticWrapper.js
import * as React from 'react';
import { withStyles } from '@material-ui/core/styles';
import { DIALOG_WIDTH } from '../constants/dimensions';
import { WrapperVariantContext, IsStaticVariantContext } from './WrapperVariantContext';
import { jsx as _jsx } from "react/jsx-runtime";

var styles = function styles(theme) {
  return {
    root: {
      overflow: 'hidden',
      minWidth: DIALOG_WIDTH,
      display: 'flex',
      flexDirection: 'column',
      backgroundColor: theme.palette.background.paper
    }
  };
};

var StaticWrapper = function StaticWrapper(props) {
  var classes = props.classes,
      _props$displayStaticW = props.displayStaticWrapperAs,
      displayStaticWrapperAs = _props$displayStaticW === void 0 ? 'mobile' : _props$displayStaticW,
      children = props.children;
  var isStatic = true;
  return /*#__PURE__*/_jsx(IsStaticVariantContext.Provider, {
    value: isStatic,
    children: /*#__PURE__*/_jsx(WrapperVariantContext.Provider, {
      value: displayStaticWrapperAs,
      children: /*#__PURE__*/_jsx("div", {
        className: classes.root,
        children: children
      })
    })
  });
};

export default withStyles(styles, {
  name: 'MuiPickersStaticWrapper'
})(StaticWrapper);
