// @material-ui/lab/modern/internal/pickers/PickersToolbarButton.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _objectWithoutPropertiesLoose from "@babel/runtime/helpers/esm/objectWithoutPropertiesLoose";
import * as React from 'react';
import clsx from 'clsx';
import Button from '@material-ui/core/Button';
import { withStyles } from '@material-ui/core/styles';
import PickersToolbarText from './PickersToolbarText';
import { jsx as _jsx } from "react/jsx-runtime";
export const styles = {
  root: {
    padding: 0,
    minWidth: '16px',
    textTransform: 'none'
  }
};

const ToolbarButton = props => {
  const {
    align,
    classes,
    className,
    selected,
    typographyClassName,
    value,
    variant
  } = props,
        other = _objectWithoutPropertiesLoose(props, ["align", "classes", "className", "selected", "typographyClassName", "value", "variant"]);

  return /*#__PURE__*/_jsx(Button, _extends({
    variant: "text",
    className: clsx(classes.root, className)
  }, other, {
    children: /*#__PURE__*/_jsx(PickersToolbarText, {
      align: align,
      className: typographyClassName,
      variant: variant,
      value: value,
      selected: selected
    })
  }));
};

export default withStyles(styles, {
  name: 'MuiPickersToolbarButton'
})(ToolbarButton);
